erpnext/accounts/doctype/mode_of_payment/mode_of_payment.py

```diff
@@ -5,6 +5,32 @@
 import frappe
 
 from frappe.model.document import Document
+from frappe import _
 
 class ModeofPayment(Document):
-	pass
+	def validate(self):
+		self.validate_accounts()
+		self.validate_repeating_companies()
+	
+	def validate_repeating_companies(self):
+		"""Error when Same Company is entered multiple times in accounts"""
+		accounts_list = []
+		for entry in self.accounts:
+			accounts_list.append(entry.company)
+
+		if len(accounts_list)!= len(set(accounts_list)):
+			frappe.throw(_("Same Company is entered more than once"))
+
+	def validate_accounts(self):
+		for entry in self.accounts:
+			"""Error when Company of Ledger account doesn't match with Company Selected"""
+			account = frappe.get_list("Account",
```

Comment:
this is simpler: `frappe.db.get_value("Account", entry.default_account, "company") != entry.company`